symfony/plugins/orangehrmAdminPlugin/Api/EducationAPI.php

```diff
@@ -0,0 +1,160 @@
+    public function getEducationService(): EducationService
+    {
+        if (is_null($this->educationService)) {
+            $this->educationService = new EducationService();
+        }
+        return $this->educationService;
+    }
+
+    /**
+     * @param EducationService $educationService
+     */
+    public function setEducationService(EducationService $educationService): void
+    {
+        $this->educationService = $educationService;
+    }
+
+    /**
+     * @return EndpointGetOneResult
+     * @throws RecordNotFoundException
+     * @throws DaoException
+     * @throws Exception
+     */
+    public function getOne(): EndpointGetOneResult
+    {
+        // TODO:: Check data group permission
+        $id = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, self::PARAMETER_ID);
+        $educations = $this->getEducationService()->getEducationById($id);
+        if (!$educations instanceof Education) {
+            throw new RecordNotFoundException('No Record Found');
+        }
+
+        return new EndpointGetOneResult(EducationModel::class, $educations);
+    }
+
+    /**
+     * @return EndpointGetAllResult
+     * @throws DaoException
+     * @throws RecordNotFoundException
+     * @throws Exception
+     */
+    public function getAll(): EndpointGetAllResult
+    {
+        // TODO:: Check data group permission
+
+        $educationParamHolder = new UserSearchFilterParams();
+        $this->setSortingAndPaginationParams($educationParamHolder);
+        $educations = $this->getEducationService()->getEducationList($educationParamHolder);
+        $count = $this->getEducationService()->getEducationCount($educationParamHolder);
+        return new EndpointGetAllResult(EducationModel::class, $educations, new ParameterBag(['total' => $count]));
+    }
+
+    /**
+     * @inheritDoc
+     * @throws Exception
+     */
+    public function create(): EndpointCreateResult
+    {
+        // TODO:: Check data group permission
+        $educations = $this->saveEducation();
+
+        return new EndpointCreateResult(EducationModel::class, $educations);
+    }
+
+    /**
+     * @inheritDoc
+     * @throws Exception
+     */
+    public function update(): EndpointUpdateResult
+    {
+        // TODO:: Check data group permission
+        $educations = $this->saveEducation();
+
+        return new EndpointUpdateResult(EducationModel::class, $educations);
+    }
+
+    /**
+     * @return Education
+     * @throws DaoException
+     */
+    public function saveEducation(): Education
+    {
+        $id = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, self::PARAMETER_ID);
+        $name = $this->getRequestParams()->getString(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_NAME);
+        if (!empty($id)) {
+            $education = $this->getEducationService()->getEducationById($id);
+        } else {
+            $education = new Education();
+        }
+
+        $education->setName($name);
```

Comment:
This allows saving empty education name

Reply:
fixed using this condition
if ($education == null){
                throw new RecordNotFoundException('No Record Found');
            }